### PR TITLE
feat: Badge support classNames & styles

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: Badge support classNames & styles |
| 🇨🇳 Chinese | feat: Badge 支持 classNames 属性和 styles 属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 27ccb12</samp>

This pull request adds new features to the Badge component, to allow users to customize its appearance with `classNames` and `styles` props. It also updates the documentation in English and Chinese, and adds a test case for the new features.

### <samp>🤖 Generated by Copilot at 27ccb12</samp>

*  Add new props `classNames` and `styles` to Badge component, to allow customizing the root and count elements ([link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391R43-R50), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391R70-R71), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391R188), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L185-R196), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L197-R208), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L213-R224), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L223-R234))
* Update Badge component documentation in English and Chinese, to explain the usage and type of the new props ([link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-c8fdf756230cf3ebcb4f347ba42136a13694db1ab8c7f95bb30bbfa27fa2ca82R44), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-c8fdf756230cf3ebcb4f347ba42136a13694db1ab8c7f95bb30bbfa27fa2ca82R51), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-46d43a0f65a205837cc26cbbec241f700f5a5a74712c02922f11584de845bed4R45), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-46d43a0f65a205837cc26cbbec241f700f5a5a74712c02922f11584de845bed4R52))
* Add a new test case for Badge component, to verify that it supports custom classNames and styles for the root and count elements ([link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-ee979b84e5ca82bb9cfeca3d077bff780d88bc5b51d2dce6a2de35c180c5b901R224-R251))
* Fix a typo in Badge component, to use the correct `classnames` function instead of `classNames` ([link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L157-R167), [link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L169-R179))
* Reorder the import statements in `components/badge/index.tsx`, to follow the alphabetical order and the convention of importing types separately ([link](https://github.com/ant-design/ant-design/pull/43245/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391L1-R10))

### CODE REVIEW

File: components/badge/index.en-US.md [hunk 1]
```diff
@@ -41,12 +41,14 @@ Badge normally appears in proximity to notifications or user avatars with eye-ca
 | --- | --- | --- | --- | --- |
 | color | Customize Badge dot color | string | - |  |
 | count | Number to show in badge | ReactNode | - |  |
+| classNames | set Badge in common className | { root?: string; count?: string } | - | 5.7.0 |
```
Comment (MadCcc) [inline components/badge/index.en-US.md:44]: 参考一下 Button，语义结构也要留个 table

Comment (li-jia-nan) [reply components/badge/index.en-US.md]: done

File: components/badge/__tests__/index.test.tsx [hunk 2]
```diff
@@ -221,4 +221,32 @@ describe('Badge', () => {
...
+    const { container } = render(
+      <Badge
+        count={10}
+        classNames={{
+          root: 'test-root',
+          count: 'test-count',
+        }}
+        styles={{
+          root: { backgroundColor: 'yellow' },
+          count: { backgroundColor: 'blue' },
```
Comment (MadCcc) [inline components/badge/__tests__/index.test.tsx:235]:
```
{
  root: string,
  indicator: string;
}

Comment (li-jia-nan) [reply components/badge/__tests__/index.test.tsx]: done

File: components/badge/index.en-US.md [hunk 3]
```diff
@@ -58,6 +60,13 @@ Badge normally appears in proximity to notifications or user avatars with eye-ca
 | placement | The placement of the Ribbon, `start` and `end` follow text direction (RTL or LTR) | `start` \| `end` | `end` |  |
 | text | Content inside the Ribbon | ReactNode | - |  |
 
+### `styles` and `classNames` attribute
+
+| Property  | Description         | Version |
+| --------- | ------------------- | ------- |
+| root      | set `root` element  | 5.5.0   |
```
Comment (MadCcc) [inline components/badge/index.en-US.md:67]:
```suggestion
| root      | set `root` element  | 5.7.0   |
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**